pkg/worker/handler/operator/registry/ensure.go

```diff
@@ -0,0 +1,117 @@
+package registry
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/0xSplits/kayron/pkg/release/artifact"
+	"github.com/0xSplits/kayron/pkg/release/schema/service"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
+	"github.com/aws/aws-sdk-go-v2/service/ecr/types"
+	"github.com/xh3b4sd/choreo/parallel"
+	"github.com/xh3b4sd/tracer"
+)
+
+func (r *Registry) Ensure() error {
+	var err error
+
+	var ser []service.Service
+	for i := range r.ser.Length() {
+		var s service.Service
+		{
+			s, _ = r.ser.Search(i)
+		}
+
+		{
+			ser = append(ser, s)
+		}
+	}
+
+	// Check whether the desired Docker image exists within the underlying
+	// container registry, if the current and desired state differs.
+
+	fnc := func(i int, x service.Service) error {
+		var err error
+
+		var cur string
+		var des string
+		{
+			cur, _ = r.art.Search(artifact.ReferenceCurrent(i))
+			des, _ = r.art.Search(artifact.ReferenceDesired(i))
+		}
+
+		// We do not have to do any work here if the currently deployed service
+		// already matches the desired service release. Note that we also have to
+		// prevent the infrastructure configuration that is part of the cached
+		// services to initiate its own image lookup, because there is no image for
+		// our templates. The current setup is not optimal and a better data
+		// structure may be used in a future refactoring, so that we do not have to
+		// do these provider checks anymore.
+
+		if cur == des || x.Provider == "cloudformation" {
+			return nil
+		}
```

Comment:
At the moment we treat everything as a service release, even the config for the cloudformation templates. We do this now because it was easy to get started, though it shows in several places already in the code that this is wrong. Right now I do not have a good idea to make this better, and so I keep focussing on the V1.